### ISSUE TITLE
Rails 5.1 deprecation warnings

### ISSUE THREAD

**Issue by doits** [magnusvk/counter_culture#162]
With the latest Rails 5.1.0.beta1 I get the following deprecation warnings:

> DEPRECATION WARNING: The behavior of `changed_attributes` inside of after callbacks will be changing in the next version of Rails. The new return value will reflect the behavior of calling the method after `save` returned (e.g. the opposite of what it returns now). To maintain the current behavior, use `saved_changes.transform_values(&:first)` instead.

> DEPRECATION WARNING: The behavior of `attribute_changed?` inside of after callbacks will be changing in the next version of Rails. The new return value will reflect the behavior of calling the method after `save` returned (e.g. the opposite of what it returns now). To maintain the current behavior, use `saved_change_to_attribute?` instead.

I traced this back to having a counter culture on my model (unfortunately I don't have an exact backtrace, but when I remove the counter the deprecation warning is gone). According to the deprecation message it looks like the "fix" is easy (just swap out some method calls?).

**Comment by doits** [comment]
Backtrace for `changed_attributes`:

```
DEPRECATION WARNING: The behavior of `changed_attributes` inside of after callbacks will be changing in the next version of Rails. The new return value will reflect the behavior of calling the method afte
r `save` returned (e.g. the opposite of what it returns now). To maintain the current behavior, use `saved_changes.transform_values(&:first)` instead. (called from block in accept_event_applications at /U
sers/markus/svn/caseman-events/app/models/event.rb:257)
/Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/attribute_methods/dirty.rb:103:in `changed_attributes'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/counter_culture-1.3.1/lib/counter_culture/counter.rb:184:in `previous_model'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/counter_culture-1.3.1/lib/counter_culture/extensions.rb:111:in `block (2 levels) in _update_counts_after_update'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/counter_culture-1.3.1/lib/counter_culture/extensions.rb:108:in `each'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/counter_culture-1.3.1/lib/counter_culture/extensions.rb:108:in `block in _update_counts_after_update'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/counter_culture-1.3.1/lib/counter_culture/extensions.rb:80:in `_wrap_in_counter_culture_active'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/counter_culture-1.3.1/lib/counter_culture/extensions.rb:107:in `_update_counts_after_update'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:413:in `block in make_lambda'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:235:in `block in halting_and_conditional'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:511:in `block in invoke_after'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:511:in `each'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:511:in `invoke_after'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:132:in `run_callbacks'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:825:in `_run_update_callbacks'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/callbacks.rb:344:in `_update_record'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/timestamp.rb:107:in `_update_record'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/persistence.rb:555:in `create_or_update'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/callbacks.rb:336:in `block in create_or_update'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:131:in `run_callbacks'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:825:in `_run_save_callbacks'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/callbacks.rb:336:in `create_or_update'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/persistence.rb:125:in `save'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/validations.rb:44:in `save'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/attribute_methods/dirty.rb:35:in `save'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/transactions.rb:308:in `block (2 levels) in save'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/transactions.rb:384:in `block in with_transaction_returning_status'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/connection_adapters/abstract/database_statements.rb:225:in `block in transaction'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/connection_adapters/abstract/transaction.rb:186:in `within_new_transaction'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/connection_adapters/abstract/database_statements.rb:225:in `transaction'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/transactions.rb:210:in `transaction'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/transactions.rb:381:in `with_transaction_returning_status'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/transactions.rb:308:in `block in save'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/transactions.rb:323:in `rollback_active_record_state!'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/transactions.rb:307:in `save'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/suppressor.rb:42:in `save'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/persistence.rb:263:in `update_attribute'
```

Backtrace for `attribute_changed?`:

```
DEPRECATION WARNING: The behavior of `attribute_changed?` inside of after callbacks will be changing in the next version of Rails. The new return value will reflect the behavior of calling the me[90/4918]
r `save` returned (e.g. the opposite of what it returns now). To maintain the current behavior, use `saved_change_to_attribute?` instead. (called from block in accept_event_applications at /Users/markus/s
vn/caseman-events/app/models/event.rb:257)
/Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/attribute_methods/dirty.rb:225:in `attribute_changed?'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activemodel-5.1.0.beta1/lib/active_model/attribute_methods.rb:384:in `user_id_changed?'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/counter_culture-1.3.1/lib/counter_culture/extensions.rb:114:in `block (2 levels) in _update_counts_after_update'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/counter_culture-1.3.1/lib/counter_culture/extensions.rb:108:in `each'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/counter_culture-1.3.1/lib/counter_culture/extensions.rb:108:in `block in _update_counts_after_update'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/counter_culture-1.3.1/lib/counter_culture/extensions.rb:80:in `_wrap_in_counter_culture_active'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/counter_culture-1.3.1/lib/counter_culture/extensions.rb:107:in `_update_counts_after_update'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:413:in `block in make_lambda'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:235:in `block in halting_and_conditional'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:511:in `block in invoke_after'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:511:in `each'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:511:in `invoke_after'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:132:in `run_callbacks'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:825:in `_run_update_callbacks'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/callbacks.rb:344:in `_update_record'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/timestamp.rb:107:in `_update_record'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/persistence.rb:555:in `create_or_update'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/callbacks.rb:336:in `block in create_or_update'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:131:in `run_callbacks'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activesupport-5.1.0.beta1/lib/active_support/callbacks.rb:825:in `_run_save_callbacks'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/callbacks.rb:336:in `create_or_update'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/persistence.rb:125:in `save'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/validations.rb:44:in `save'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/attribute_methods/dirty.rb:35:in `save'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/transactions.rb:308:in `block (2 levels) in save'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/transactions.rb:384:in `block in with_transaction_returning_status'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/connection_adapters/abstract/database_statements.rb:225:in `block in transaction'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/connection_adapters/abstract/transaction.rb:186:in `within_new_transaction'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/connection_adapters/abstract/database_statements.rb:225:in `transaction'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/transactions.rb:210:in `transaction'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/transactions.rb:381:in `with_transaction_returning_status'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/transactions.rb:308:in `block in save'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/transactions.rb:323:in `rollback_active_record_state!'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/transactions.rb:307:in `save'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/suppressor.rb:42:in `save'
  /Users/markus/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/activerecord-5.1.0.beta1/lib/active_record/persistence.rb:263:in `update_attribute'
```

**Comment by magnusvk** [comment]
Yeah I've been working on this. It's not super straightforward, unfortunately. But, hey, Rails 5.1 is only at its first beta, so I've got time, right? 😄

**Comment by doits** [comment]
Sure, thanks for the update, just wanted to make sure all the info you need is here :-)

**Comment by magnusvk** [comment]
@doits this should be fixed in the 1.4.0 gem version.
